tsl/test/sql/cagg_query.sql

```diff
@@ -296,3 +296,641 @@ ORDER BY m1.location COLLATE "C" NULLS LAST, m1.timec DESC NULLS LAST, firsth NU
+  time timestamptz NOT NULL,
+  value float
+);
+
+SELECT create_hypertable('temperature', 'time');
+
+INSERT INTO temperature VALUES ('2000-01-01 01:00:00'::timestamptz, 5);
+
+CREATE TABLE temperature_wo_tz (
+  time timestamp NOT NULL,
+  value float
+);
+
+SELECT create_hypertable('temperature_wo_tz', 'time');
+
+INSERT INTO temperature_wo_tz VALUES ('2000-01-01 01:00:00'::timestamp, 5);
+
+CREATE TABLE temperature_date (
+  time date NOT NULL,
+  value float
+);
+
+SELECT create_hypertable('temperature_date', 'time');
+
+INSERT INTO temperature_date VALUES ('2000-01-01 01:00:00'::timestamp, 5);
+
+-- Integer based tables
+CREATE TABLE table_smallint (
+  time smallint,
+  data smallint
+);
+
+CREATE TABLE table_int (
+  time int,
+  data int
+);
+
+CREATE TABLE table_bigint (
+  time bigint,
+  data bigint
+);
+
+SELECT create_hypertable('table_smallint', 'time', chunk_time_interval => 10);
+SELECT create_hypertable('table_int', 'time', chunk_time_interval => 10);
+SELECT create_hypertable('table_bigint', 'time', chunk_time_interval => 10);
+
+CREATE OR REPLACE FUNCTION integer_now_smallint() returns smallint LANGUAGE SQL STABLE as $$ SELECT coalesce(max(time), 0) FROM table_smallint $$;
+CREATE OR REPLACE FUNCTION integer_now_int() returns int LANGUAGE SQL STABLE as $$ SELECT coalesce(max(time), 0) FROM table_int $$;
+CREATE OR REPLACE FUNCTION integer_now_bigint() returns bigint LANGUAGE SQL STABLE as $$ SELECT coalesce(max(time), 0) FROM table_bigint $$;
+
+SELECT set_integer_now_func('table_smallint', 'integer_now_smallint');
+SELECT set_integer_now_func('table_int', 'integer_now_int');
+SELECT set_integer_now_func('table_bigint', 'integer_now_bigint');
+
+INSERT INTO table_smallint VALUES(1,2);
+INSERT INTO table_int VALUES(1,2);
+INSERT INTO table_bigint VALUES(1,2);
+
+---
+-- Tests with CAgg creation
+---
+CREATE MATERIALIZED VIEW cagg_4_hours
+  WITH (timescaledb.continuous, timescaledb.materialized_only = false) AS
+  SELECT time_bucket('4 hour', time), max(value)
+    FROM temperature
+    GROUP BY 1 ORDER BY 1;
+SELECT * FROM _timescaledb_catalog.continuous_aggs_bucket_function ORDER BY 1 DESC LIMIT 1;
+DROP MATERIALIZED VIEW cagg_4_hours;
+
+CREATE MATERIALIZED VIEW cagg_4_hours_offset
+  WITH (timescaledb.continuous, timescaledb.materialized_only = false) AS
+  SELECT time_bucket('4 hour', time, '30m'::interval), max(value)
+    FROM temperature
+    GROUP BY 1 ORDER BY 1;
+SELECT * FROM _timescaledb_catalog.continuous_aggs_bucket_function ORDER BY 1 DESC LIMIT 1;
+DROP MATERIALIZED VIEW cagg_4_hours_offset;
+
+CREATE MATERIALIZED VIEW cagg_4_hours_offset2
+  WITH (timescaledb.continuous, timescaledb.materialized_only = false) AS
+  SELECT time_bucket('4 hour', time, "offset"=>'30m'::interval), max(value)
+    FROM temperature
+    GROUP BY 1 ORDER BY 1;
+SELECT * FROM _timescaledb_catalog.continuous_aggs_bucket_function ORDER BY 1 DESC LIMIT 1;
+DROP MATERIALIZED VIEW cagg_4_hours_offset2;
+
+\set ON_ERROR_STOP 0
```

Comment:
One think that I like to see on tests when it should fail is a small comment about the failure... I understand everything is happening here but people that are not familiar with some code can understand it better by reading the tests. Make sense for you???

Reply:
Added some comments on why the CAgg definitions are not valid.